shielder/deploy/deploy.sh

```diff
@@ -145,6 +153,15 @@ docker_cargo() {
     -c "cargo ${1}"
 }
 
+ci_docker_cargo() {
```

Comment:
We need a different version of the method when running in CI due to issues w/ permissions when running as a docker process - we can't create directories in cargo cache (and we try to do that when compiling contracts). Hence mounting of these dirs is removed.

Reply:
I find that hard to believe - since it works just fine for compiling [here](https://github.com/Cardinal-Cryptography/aleph-node/actions/runs/4502925871/jobs/7925439818#step:11:22)

Reply:
In TheButton you're copying everything and running Makefile inside docker.

Reply:
> Take a good look at mapping users and user groups when running these containers, that's usually what is responsible for no read/write privileges.

I had tried that, asked on Slack last week as well, but it doesn't work. Docker user is probably not a part of the required group on these machines and is not allowed to write to `~/.cargo/*` dirs.

Reply:
Then I don't understand why is the deployment script set up differently than in TheButton? It runs CI on exactly the same self-hosted instance. It mounts and writes to the contract directories no problem.

Either you don't have to mount .cargo, or just ask devops guys to make it writable for the usergroup that the docker is part of. Or launch docker as a user in one of those groups. We control that instance.

Having two methods to launch will come back and bite, and will make any problems on CI irreproducible locally, making it even harder to debug potential problems.

Reply:
In TheButton you don't have the self-contained Makefile and CI is set up differently from the local development. Here I'm re-using the same scripts, commands, etc _except_ for these 3 lines in `cargo-docker`.

> Having two methods to launch will come back and bite, and will make any problems on CI irreproducible locally, making it even harder to debug potential problems.

This setup is IMO actually going to help w/ that - the local vs CI setup is almost identical (again, except for the part whether you use cached dependencies to download them every time). It's much better than the situation TheButton in that case.